src/Core/Auth/Identity/AuthenticatorTokenProvider.cs

```diff
@@ -2,18 +2,31 @@
 using Bit.Core.Entities;
 using Bit.Core.Services;
 using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Caching.Distributed;
 using Microsoft.Extensions.DependencyInjection;
 using OtpNet;
 
 namespace Bit.Core.Auth.Identity;
 
 public class AuthenticatorTokenProvider : IUserTwoFactorTokenProvider<User>
 {
+    private const string CacheKeyFormat = "Authenticator_TOTP_{0}_{1}";
+
     private readonly IServiceProvider _serviceProvider;
+    private readonly IDistributedCache _distributedCache;
+    private readonly DistributedCacheEntryOptions _distributedCacheEntryOptions;
 
-    public AuthenticatorTokenProvider(IServiceProvider serviceProvider)
+    public AuthenticatorTokenProvider(
+        IServiceProvider serviceProvider,
+        [FromKeyedServices("persistent")]
+        IDistributedCache distributedCache)
     {
         _serviceProvider = serviceProvider;
+        _distributedCache = distributedCache;
+        _distributedCacheEntryOptions = new DistributedCacheEntryOptions
+        {
+            AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(5)
```

Comment:
❓ Where's this value coming from?

Reply:
I just made it up. 10x the time window of the TOTP code. Authenticator window is 30 seconds

Reply:
Isn't the point of this cache to expect invalidation via TTL though? That's what enforces the replay protection.

Reply:
Yes, but I think you have it backwards. The cache here only exists once the code is used, to prevent the replay. So it needs to cover the entirety of the time the code would else by valid for. So I just added some cushion around that window.

Reply:
I guess I was thinking that the TTL should match how long a code is valid for. Why go any longer?

Reply:
I am not exactly sure how much padding exists on the side of the available time windows, so I just made the TTL long enough to make sure it encompassed it entirely.